proto/redpanda/api/dataplane/v1alpha1/kafka_connect.proto

```diff
@@ -0,0 +1,328 @@
+  ];
+  string cluster_name = 2 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+
+  option (buf.validate.message).cel = {
+    id: "expand_can_only_be_status_or_info",
+    message: "Valid expand values are: info, status",
+    expression: "this.expand.all(x, x in [\"info\", \"status\"])"
+  };
+
+  // Value of the next_page_token field returned by the previous response. If not provided, the system assumes the first page is requested.
+  string page_token = 3;
+}
+
+message RestartConnectorRequest {
+  message Options {
+    bool include_tasks = 1;
+    bool only_failed = 2;
+  }
+  string cluster_name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+  string name = 2 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+
+  Options options = 3;
+}
+
+message DeleteConnectorRequest {
+  string cluster_name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+  string name = 2 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+}
+
+message PauseConnectorRequest {
+  string cluster_name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+  string name = 2 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+}
+
+message ResumeConnectorRequest {
+  string cluster_name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+  string name = 2 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+}
+
+message GetConnectorRequest {
+  string cluster_name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+  string name = 2 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+}
+
+message CreateConnectorRequest {
+  string cluster_name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+  ConnectorSpec connector = 2;
+}
+
+message GetConnectorResponse {
+  ConnectorSpec connector = 1;
+  HolisticState holistic_state = 4;
+  repeated ConnectorError errors = 5;
+}
+
+message CreateConnectorResponse {
+  string name = 1;
+  // cod fig is the connector cron configuration map, every key is a string
+  map<string, string> config = 2;
+  repeated TaskInfo tasks = 3 [(google.api.field_behavior) = OUTPUT_ONLY];
+  string type = 4 [(google.api.field_behavior) = OUTPUT_ONLY];
+}
+
+message ListConnectorsResponse {
+  // connectors is the list of connectors the key is the connector name
+  repeated ConnectorInfoStatus connectors = 1;
+  message ConnectorInfoStatus {
```

Comment:
Can we move this at the top of this struct? I prefer to define types first before they are being used. Maybe opinionated if not already enforced by the linter :D